### PR TITLE
feat: k8s custom port expose

### PR DESCRIPTION
**Pull request by florian-hoenicke**
Currently, the only possible port to reach the k8s gateway is 8080.
This pr makes it possible to specify the port number in the flow

### CODE REVIEW

File: cli/autocomplete.py [hunk 1]
```diff
@@ -189,8 +189,24 @@
             '--pea-id',
             '--pea-role',
             '--noblock-on-start',
-            '--dynamic-routing',
+            '--uses-before',
```
Comment (JoanFM) [inline cli/autocomplete.py:192]: This  is wrong, no uses before for gateway. Or parallel , shards and more

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/parsers/__init__.py [hunk 2]
```diff
@@ -89,6 +89,8 @@ def set_gateway_parser(parser=None):
     mixin_comm_protocol_parser(parser)
     mixin_gateway_parser(parser)
     mixin_pea_parser(parser)
+    mixin_base_pod_parser(parser)
```
Comment (JoanFM) [inline jina/parsers/__init__.py:92]: this mixin base should not be added I think

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/parsers/__init__.py (hunk 2 above)
Comment (florian-hoenicke) [reply jina/parsers/__init__.py:92]: What is the reason it should not be added? It is a gateway pod and would be the first step towards scaling the gateway.

File: cli/autocomplete.py (hunk 1 above)
Comment (florian-hoenicke) [reply cli/autocomplete.py:192]: If we say the gateway is a pod, it should have all features of a pod

Comment (JoanFM) [reply cli/autocomplete.py:192]: Right now this is not true, when this is true it may happen. But not now.

Comment (florian-hoenicke) [reply cli/autocomplete.py:192]: true, I changed it. Thanks

File: jina/parsers/__init__.py (hunk 2 above)
Comment (florian-hoenicke) [reply jina/parsers/__init__.py]: changed it since we don't scale the gateway for now

**Review by JoanFM — APPROVED** [review]

**Pull request merged**